lib/syskit/network_generation/system_network_generator.rb

```diff
@@ -399,39 +483,52 @@ def self.verify_all_deployments_are_unique(
 
                 return if using_same_deployment.empty?
 
-                raise ConflictingDeploymentAllocation.new(
-                    using_same_deployment, toplevel_tasks_to_requirements
-                ), "there are deployments used multiple times"
+                using_same_deployment.each do |orocos_name, tasks|
+                    exception = ConflictingDeploymentAllocation.new(
+                        orocos_name, tasks, toplevel_tasks_to_requirements
+                    )
+                    error_handler.register_resolution_failures_from_exception(
+                        tasks, exception, "deployment used multiple times"
+                    )
+                end
             end
 
             # Validates the network generated by {#compute_system_network}
             #
             # It performs the tests that are only needed on an abstract network,
             # i.e. on a network in which some tasks are still abstract
-            def validate_abstract_network
+            def validate_abstract_network(error_handler: @error_handler)
                 self.class.verify_no_multiplexing_connections(plan)
```

Comment:
Not passing error_handler to that one ?

Reply:
For the record: neither the composition model creation, nor the merge solver, should allow the creation of multiplexing connections. So, indeed, sanity checl.